dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java

```diff
@@ -163,9 +166,10 @@ private LinkageChecker(
             }
           }
         }
-        if (!classFile.getClassPathEntry().getClassNames()
-            .contains(classSymbol.getClassBinaryName())) {
 
+        String classBinaryName = classSymbol.getClassBinaryName();
```

Comment:
Avoid moving data between classes. Instead ClassSymbol should be able to return the classFileName if that's what we need.

Reply:
A ClassSymbol cannot know the class file name. A symbol is an abstract entity apart from concrete class files. For example, when a class file's constant pool references a class symbol of "com.google.Foo", it does not know the class symbol is materialized by a class file "BOOT-INF/classes/com/google/Foo.class", another class file "WEB-INF/classes/com/google/Foo.class", or just "com/google/Foo.class".